tempesta_fw/t/unit/test_tfw_str.c

```diff
@@ -178,6 +183,95 @@ TEST(cstr, simd_match_ctext_vchar)
 #undef __S
 }
 
+TEST(cstr, simd_match_custom)
+{
+	int i;
+	unsigned char a[256] = {};
+
+	for (i = 0; i < 256; ++i) {
+		if (i == 0x21
+		    || (0x23 <= i && i <= 0x3B)
+		    || i == 0x3D
+		    || (0x3f <= i && i <= 0x5a)
+		    || (0x61 <= i && i <= 0x7A)
+		    || i == 0x5b || i == 0x5d || i == 0x5f
+		    || i == 0x7E || i == 0x98)
+			a[i] = 1;
+	}
+
+	/*
+	 * Set custom alphabet to URI character set plus x98, but use xff
+	 * matcher just because it's different from URI and CTEXT+VCHAR to
+	 * enforce custom matching.
+	 */
+	tfw_init_custom_xff(a, true);
```

Comment:
Custom `xff` alphabet is set here but not disabled after the test, so tests from `test_http_parser.c` (which started after `tfw_str` tests) with `X-Forwarded-For` header - are failed.